### ISSUE TITLE
data.azurerm_client_config.current.object_id return "" empty when upgrading az cli to 2.37.0

### ISSUE THREAD

**Issue by LaurentLesle** [hashicorp/terraform-provider-azurerm#16982]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.1

### AzureRM Provider Version

3.7.0 or 2.88.1

### Affected Resource(s)/Data Source(s)

data.azurerm_client_config.current.object_id

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

data "azurerm_client_config" "current" {}

output "object_id" {
  value = data.azurerm_client_config.current.object_id
}
```

### Debug Output/Panic Output

```shell
data.azurerm_client_config.current: Reading...
data.azurerm_client_config.current: Read complete after 0s [id=2022-05-27 01:16:03.753374069 +0000 UTC]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:
object_id = ""
```

### Expected Behaviour

object_id = "object id of the logged in user"

### Actual Behaviour

When upgrading the azure cli to version 2.37.0 

### Steps to Reproduce

Upgrading the azure cli to version 2.37.0 
Login to azure with a user
 terraform apply the code above

Note:
az ad signed-in-user show 
was returning objectId with previous versions
now return "id" for the user object_id

**Comment by pauldotyu** [comment]
Switching to using the [azuread](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/data-sources/client_config) provider is a workaround for now.

**Comment by geekzter** [comment]
This also works:

```
data external account_info {
  program                      = [
                                 "az",
                                 "ad",
                                 "signed-in-user",
                                 "show",
                                 "--query",
                                 "{object_id:id}",
                                 "-o",
                                 "json",
                                 ]
}

output user_object_id {
    value = data.external.account_info.result.object_id
}
```

**Comment by tombuildsstuff** [comment]
Fixed via #17074
